quipucords/scanner/acs/inspect.py

```diff
@@ -0,0 +1,156 @@
+"""Inspection phase for ACS scanner."""
+
+from __future__ import annotations
+
+from logging import getLogger
+
+from django.conf import settings
+from django.db import transaction
+from requests import RequestException
+
+from api.models import RawFact, ScanTask, SystemInspectionResult
+from scanner.acs.runner import AcsTaskRunner
+from scanner.exceptions import ScanFailureError
+
+logger = getLogger(__name__)
+
+
+class InspectTaskRunner(AcsTaskRunner):
+    """Inspection phase task runner for ACS scanner."""
+
+    supports_partial_results = False
+
+    REQUEST_KWARGS = {
+        "raise_for_status": True,
+        "timeout": settings.QPC_INSPECT_TASK_TIMEOUT,
+    }
+
+    def execute_task(self, manager_interrupt):
+        """
+        Execute the task and save the results.
+
+        :param manager_interrupt: interrupt that can pause/cancel the scan
+        :returns: tuple of human readable message and ScanTask.STATUS_CHOICE
+        """
+        self._check_prerequisites()
+
+        results = {}
+        inspection_status = SystemInspectionResult.SUCCESS
+        collectable_facts = ("secured_units_current", "secured_units_max")
+        for fact in collectable_facts:
+            method = getattr(self, f"get_{fact}")
+            try:
+                results[fact] = method()
+            except RequestException:
+                logger.exception(
+                    "Error collecting fact '%s' for ACS host '%s'.",
+                    fact,
+                    self.system_name,
+                )
+                inspection_status = SystemInspectionResult.FAILED
+
+        inspection_status = SystemInspectionResult.SUCCESS
```

Comment:
the logic here is borked. Line 52 is overriding 50. You can probably safely remove line 52.